learn/core_concepts/documents.md

```diff
@@ -124,3 +124,19 @@ Since CSV does not support arrays or nested objects, `cast` cannot be converted
 ::: note
 If you don't specify the data type for an attribute, it will default to `:string`.
 :::
+
+### Auto-batching
+
+Auto-batching combines consecutive document addition requests into a single batch and processes them together. This significantly speeds up the indexing process.
+
+Meilisearch batches document addition requests when they:
+
+- Target the same index
+- Have the same update method (i.e., [POST](/reference/api/documents.md#add-or-replace-documents) or [PUT](/reference/api/documents.md#add-or-update-documents))
+- Are immediately consecutive
+
+Tasks within the same batch share the same values for `startedAt`, `finishedAt`, and `duration`. 
+
+If a task fails due to an invalid document, it will be removed from the batch. The rest of the batch will still process normally. If an [internal](/reference/api/overview.md#errors) error occurs, the whole batch will fail and all tasks within it will share the same `error` object.
```

Comment:
```suggestion
If a task fails due to an invalid document, it will be removed from the batch. The rest of the batch will still process normally. If an [`internal`](/reference/api/overview.md#errors) error occurs, the whole batch will fail and all tasks within it will share the same `error` object.
```